tests/testthat/error/test-type-data-frame.txt

```diff
@@ -0,0 +1,14 @@
+
+combining data frames with foreign classes uses fallback
+========================================================
+
+> foo <- structure(mtcars[1:3], class = c("foo", "data.frame"))
+> bar <- structure(mtcars[9:11], class = c("bar", "data.frame"))
+> vec_ptype_common(foo, bar)
```

Comment:
Do you want to have a `vec_ptype_common()` here just to check that the warning only occurs once?

Reply:
yup, also improved the logic so that repeat warnings for new classes that we haven't seen before are issued.